app/code/Magento/Inventory/Model/ReservationCleanup.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Setup\Operation\CreateReservationTable;
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+
+/**
+ * @inheritdoc
+ */
+class ReservationCleanup implements ReservationCleanupInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @param ResourceConnection $resource
+     */
+    public function __construct(
+        ResourceConnection $resource
+    ) {
+        $this->resource = $resource;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $connection = $this->resource->getConnection();
+        $reservationTable = $connection->getTableName(CreateReservationTable::TABLE_NAME_RESERVATION);
+
+        $select = $connection->select()
+            ->from(
+                $reservationTable,
+                ['grouped_reservation_ids' => 'group_concat(' . ReservationInterface::RESERVATION_ID . ')']
+            )
+            ->group([ReservationInterface::STOCK_ID, ReservationInterface::SKU])
+            ->having('sum(' . ReservationInterface::QUANTITY . ') = 0');
```

Comment:
It's not the most effective way of cleaning up reservations because we will remove all reservations (for given stock_id and sku) just in case - all the reservations quantities summing up will result **0**

For example, if we have 
```
SKU   STOCK_ID  Reservation
sku-1     1     -10
sku-1     1      5
sku-1     1      2
sku-1     1      3
sku-1     1      -1
```
we will leave it as-is, but potentially already in this case we could remove everything except 
`sku-1     1      -1`

But I agree that currently we could leave it as-is